lib/menubar/utils.lua

```diff
@@ -243,17 +246,49 @@ end
 
 --- Parse a directory with .desktop files recursively.
 -- @tparam string dir The directory.
--- @treturn table Paths of found .desktop files.
-function utils.parse_dir(dir)
-    local programs = {}
-    local files = io.popen('find '.. dir .." -name '*.desktop' 2>/dev/null")
-    for file in files:lines() do
-        local program = utils.parse_desktop_file(file)
-        if program then
-            table.insert(programs, program)
+-- @tparam function callback Will be fired when all the files will be parsed.
```

Comment:
Same comment as before: *were* parsed and it might be nice to explain the argument a bit (or at least mention it)